shared/src/commonMain/kotlin/io/customer/shared/di/SDKComponent.kt

```diff
@@ -10,4 +12,6 @@ import io.customer.shared.sdk.config.CustomerIOConfig
 interface SDKComponent {
     val platform: Platform
     val customerIOConfig: CustomerIOConfig
+    val userAgentStore: UserAgentStore
+    val customAttributeSerializer: CustomAttributeSerializer?
```

Comment:
I'm not sure if it's the best way to get custom serializer from client apps, but with current architecture, it looked best to me as our SDKs will be wrapping this up for client apps.

For details on the class, please see my comments on it.

Reply:
I think we should consider removing the user-agent SDK wrapper code from this PR and doing that in another PR. Wait on it until we get KMM a bit further along will help us determine how best to implement this.

Reply:
I think removing it is not a good idea, as we won't be able to run KMM end-to-end without it and will still need to add some snippet to make our requests work. So I would suggest to keep it there, and address this in our pending items ticket; as once we have all the code merged, we can try different ideas to make it work and will be easy to discuss challenges coming to move it to some other place.